### PR TITLE
Feature/log4j2

### PR DESCRIPTION
**Pull request by jmichelberger**
Update to log4j2 2.9. This is only the change of the library and its usage. No further logging changes are done. Changed configuration xmls in sub projects.

### CODE REVIEW

**Comment by jmichelberger** [comment]
Build breaks because:No processor claimed any of these annotations: com.sun.javafx.beans.IDProperty.
Treat warnings as errors setting breaks the build. Local builds without that setting are working. Travis is picking jdk 8u144. Is this a new issue in that JDK? It seems it has nothing to do with user code changes.

File: build.gradle [hunk 1]
```diff
@@ -55,7 +55,8 @@ subprojects {
 
 project (":TweetWallFX-Configuration") {
     dependencies {
-        compile group: "log4j", name: "log4j", version: "1.2.17"
+        compile group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.9.0'
+        compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.9.0'
```
Comment (mklaehn) [inline build.gradle]: log4j-core should be a runtime dependency

File: build.gradle [hunk 2]
```diff
@@ -78,7 +80,8 @@ project(":TweetWallFX-Tweet-Impl-Twitter4J") {
 project(":TweetWallFX-Core"){
     dependencies {
         compile group: "de.jensd", name: "fontawesomefx", version: "8.9"
-        compile group: "log4j", name: "log4j", version: "1.2.17"
+        compile group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.9.0'
+        compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.9.0'
```
Comment (mklaehn) [inline build.gradle]: log4j-core should be a runtime dependency

File: build.gradle [hunk 3]
```diff
@@ -68,7 +69,8 @@ project(":TweetWallFX-Tweet-Impl-Twitter4J") {
     dependencies {
         compile group: "org.twitter4j", name: "twitter4j-core", version: "4.0.6"
         compile group: "org.twitter4j", name: "twitter4j-stream", version: "4.0.6"
-        compile group: "log4j", name: "log4j", version: "1.2.17"
+        compile group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.9.0'
+        compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.9.0'
```
Comment (mklaehn) [inline build.gradle]: log4j-core should be a runtime dependency

File: controls/src/main/java/org/tweetwallfx/controls/steps/AddTweetToCloudStep.java [hunk 4]
```diff
@@ -26,8 +26,9 @@
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
-import org.apache.log4j.LogManager;
-import org.apache.log4j.Logger;
+import javafx.application.Platform;
```
Comment (mklaehn) [inline controls/src/main/java/org/tweetwallfx/controls/steps/AddTweetToCloudStep.java]: unused import? If so it should be removed.

File: controls/src/main/java/org/tweetwallfx/controls/steps/CloudToTweetStep.java [hunk 5]
```diff
@@ -26,7 +26,8 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import org.apache.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
```
Comment (mklaehn) [inline controls/src/main/java/org/tweetwallfx/controls/steps/CloudToTweetStep.java]: Import is unused. please remove

File: controls/src/main/java/org/tweetwallfx/controls/steps/FadeInCloudStep.java [hunk 6]
```diff
@@ -35,7 +35,9 @@
 import javafx.geometry.Bounds;
 import javafx.scene.text.Text;
 import javafx.util.Duration;
-import org.apache.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.tweetwallfx.controls.TweetLayout;
```
Comment (mklaehn) [inline controls/src/main/java/org/tweetwallfx/controls/steps/FadeInCloudStep.java]: import is unused. please remove

**Pull request merged**